packages/openapi-generator/src/file-serializer/schema.ts

```diff
@@ -32,13 +32,18 @@ export function serializeSchema(schema: OpenApiSchema): string {
     const type = serializeSchema(schema.items);
     return schema.uniqueItems
       ? `Set<${type}>`
-      : 'properties' in schema.items
+      : 'properties' in schema.items || 'allOf' in schema.items
         ? `(${type})[]`
         : `${type}[]`;
   }
 
   if (isObjectSchema(schema)) {
-    return serializeObjectSchema(schema);
+    const types: string[] = [];
+    if (isAllOfSchema(schema)) {
+      types.push(schema.allOf.map(type => serializeSchema(type)).join(' & '));
+    }
+    types.push(serializeObjectSchema(schema));
+    return types.join(' & ');
```

Comment:
This is duplicate code, as we do this for the normal case, where we only have `allOf` as well. Also, we need to consider `oneOf` and `anyOf`.
I think it would be simpler to do this at the time of parsing instead of serializing (see my comment in the parser).